ui/src/app/shared/workflow-actions.ts

```diff
@@ -0,0 +1,64 @@
+import {NodePhase, Workflow} from '../../models';
+import {services} from './services';
+import {Utils} from './utils';
+
+export type ActionDisabled = {
+    [action in WorkflowActionName]: boolean;
+};
+
+export type WorkflowActionName = 'RETRY' | 'RESUBMIT' | 'SUSPEND' | 'RESUME' | 'STOP' | 'TERMINATE' | 'DELETE';
+
+export interface WorkflowAction {
+    title: string;
+    action: () => Promise<any>;
+    iconClassName: string;
+    disabled: (wf: Workflow) => boolean;
+}
+
+export const WorkflowActions = {
+    RETRY: {
+        title: 'RETRY',
+        iconClassName: 'fa fa-undo',
+        disabled: (wf: Workflow) => {
+            const workflowPhase: NodePhase = wf && wf.status ? wf.status.phase : undefined;
+            return workflowPhase === undefined || !(workflowPhase === 'Failed' || workflowPhase === 'Error');
+        },
+        action: services.workflows.retry
+    },
+    RESUBMIT: {
+        title: 'RESUBMIT',
+        iconClassName: 'fa fa-plus-circle',
+        disabled: () => false,
+        action: services.workflows.resubmit
+    },
+    SUSPEND: {
+        title: 'SUSPEND',
+        iconClassName: 'fa fa-pause',
+        disabled: (wf: Workflow) => !Utils.isWorkflowRunning(wf) || Utils.isWorkflowSuspended(wf),
+        action: services.workflows.suspend
+    },
+    RESUME: {
+        title: 'RESUME',
+        iconClassName: 'fa fa-play',
+        disabled: (wf: Workflow) => !Utils.isWorkflowSuspended(wf),
+        action: services.workflows.resume
+    },
+    STOP: {
+        title: 'STOP',
+        iconClassName: 'fa fa-stop-circle',
+        disabled: (wf: Workflow) => !Utils.isWorkflowSuspended(wf),
+        action: services.workflows.stop
+    },
+    TERMINATE: {
+        title: 'TERMINATE',
+        iconClassName: 'fa fa-times-circle',
+        disabled: (wf: Workflow) => !Utils.isWorkflowSuspended(wf),
+        action: services.workflows.terminate
+    },
+    DELETE: {
```

Comment:
can we make sure that we get an "are you sure" confirmation for delete?

Reply:
In the workflows toolbar and the workflows details, there's a confirmation before deleting - in fact for the workflows toolbar, there's a confirmation before performing any batch action. Would it be better to have the confirmation happen in `workflow-actions.ts`?